src/components/views/dialogs/CreateGroupDialog.js

```diff
@@ -0,0 +1,196 @@
+            this.props.onFinished(true);
+        }).catch((e) => {
+            this.setState({createError: e});
+        }).finally(() => {
+            this.setState({creating: false});
+        }).done();
+    },
+
+    _onCancel: function() {
+        this.props.onFinished(false);
+    },
+
+    /**
+     * Parse a string that may be a group ID
+     * If the string is a valid group ID, return a list of [localpart, domain],
+     * otherwise return null.
+     *
+     * @param {string} groupId The ID of the group
+     * @return {string[]} array of localpart, domain
+     */
+    _parseGroupId: function(groupId) {
+        const matches = GROUP_REGEX.exec(this.state.groupId);
+        if (!matches || matches.length < 3) {
+            return null;
+        }
+        return [matches[1], matches[2]];
+    },
+
+    render: function() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const Loader = sdk.getComponent("elements.Spinner");
+
+        if (this.state.creating) {
+            return <Loader />;
+        }
+
+        let createErrorNode;
+        if (this.state.createError) {
+            // XXX: We should catch errcodes and give sensible i18ned messages for them,
+            // rather than displaying what the server gives us, but synapse doesn't give
+            // any yet.
+            createErrorNode = <div className="error">
+                <div>{_t('Room creation failed')}</div>
+                <div>{this.state.createError.message}</div>
+            </div>;
+        }
+
+        return (
+            <BaseDialog className="mx_CreateGroupDialog" onFinished={this.props.onFinished}
+                onEnterPressed={this._onFormSubmit}
+                title={_t('Create Group')}
+            >
+                <form onSubmit={this._onFormSubmit}>
+                    <div className="mx_Dialog_content">
+                        <div className="mx_CreateGroupDialog_label">
+                            <label htmlFor="groupname">{_t('Group Name')}</label>
+                        </div>
+                        <div>
+                            <input id="groupname" className="mx_CreateGroupDialog_input"
+                                autoFocus={true} size="64"
+                                placeholder={_t('Example')}
+                                onChange={this._onGroupNameChange}
+                                value={this.state.groupName}
+                            />
+                        </div>
+                        <br />
+                        <div className="mx_CreateGroupDialog_label">
+                            <label htmlFor="groupid">{_t('Group ID')}</label>
+                        </div>
+                        <div>
+                            <input id="groupid" className="mx_CreateGroupDialog_input"
+                                size="64"
+                                placeholder={_t('+example:%(domain)s', {domain: MatrixClientPeg.get().getDomain()})}
```

Comment:
hmm, not sure - would imply it is something that's designed to change, which it isn't really. I'd like to avoid state where possible